packages/react/src/ImageSlider/ImageSlider.tsx

```diff
@@ -69,19 +86,64 @@ export const ImageSliderRoot = forwardRef(
     useEffect(() => {
       const observer = new IntersectionObserver(inView, observerOptions)
 
+      const handleKeyDown = (event: KeyboardEvent) => {
+        switch (event.key) {
+          case 'ArrowLeft':
+            goToPreviousSlide()
+            break
+          case 'ArrowRight':
+            goToNextSlide()
+            break
+        }
+      }
+
       if (targetRef.current) {
         const slides = targetRef.current.children
         const slidesArray = Array.from(slides)
         for (let slide of slidesArray) observer.observe(slide)
 
-        targetRef.current.addEventListener('scrollend', synchronise.bind(targetRef.current))
+        targetRef.current.addEventListener('scrollend', synchronise)
+        targetRef.current.addEventListener('keydown', handleKeyDown)
+      }
+
+      return () => {
+        if (targetRef.current) {
+          targetRef.current.removeEventListener('scrollend', synchronise)
+          targetRef.current.removeEventListener('keydown', handleKeyDown)
+        }
       }
     }, [targetRef, observerOptions])
 
     const synchronise = () => {
       updateControls()
     }
 
+    const handleThumbsKeyDown = (event: ReactKeyboardEvent<HTMLElement>) => {
+      const target = event.target as HTMLElement
+      const element = target.parentElement?.children[currentSlideId]
+
+      if (event.key === 'ArrowRight') {
+        const next = element?.nextElementSibling as HTMLElement | null
```

Comment:
If we give all element variables a suffix of `El` or `Element,` the code will be easier to read. (Lots of occurrences.)